mxcubeweb/core/components/lims.py

```diff
@@ -323,3 +323,30 @@ def synch_with_lims(self, lims_name):
                 self.sample_list_sync_sample(sample_info)
 
         return self.sample_list_get()
+
+    def get_lims_samples(self, lims_name: str) -> dict:
+        """
+        Get samples from LIMS.
+        This method synchronizes the sample list with LIMS and returns
+        the updated sample list.
+        :param lims_name: Name of the LIMS to synchronize with.
+        :return: Updated sample list.
+        :rtype: dict
+        """
+        self.synch_with_lims(lims_name)
+        new_sample_list = {"sampleList": {}, "sampleOrder": []}
+
+        try:
+            for loc, data in self.app.SAMPLE_LIST["sampleList"].items():
+                # If the sample has a limsID, it means it is a valid LIMS sample
+                # and should be added to the new only lims sample list
+                if data.get("limsID"):
+                    new_sample_list["sampleList"][loc] = data
+                    new_sample_list["sampleOrder"].append(loc)
+        except Exception as e:
+            logging.getLogger("MX3.HWR").info("Error while getting LIMS samples: %s", e)
```

Comment:
Why not show the whole stack trace here?
```suggestion
            logging.getLogger("MX3.HWR").exception("Error while getting LIMS samples")
```
If the `try` block was one single statement, or if we were catching specific exceptions, maybe it would be acceptable like this. But here we have a whole bunch of code in the `try` block (including a loop!), and we catch all exceptions(!). So we have very little control over what could possibly be raised and caught. Having a stack trace seems the bare minimum to me.

Reply:
Well spotted :)